### PR TITLE
Azure - Tag filters

### PR DESCRIPTION
**Pull request by tbarlow12**
This resolves issue #2502

### CODE REVIEW

File: c7n/filters/core.py [hunk 1]
```diff
@@ -406,10 +406,17 @@ def get_resource_value(self, k, i):
...
+                for t in i.get("Tags", []):
+                    if t.get('Key') == tk:
+                        r = t.get('Value')
+                        break
+            elif 'tags' in i:
+                tags = i.get("tags", [])
+                for key in tags:
+                    if key == tk:
+                        r = tags[key]
+                        break
```
Comment (stefangordon) [inline c7n/filters/core.py]: I think you can do this more concise, perhaps
```
            elif 'tags' in i:
                r = i.get("tags", {}).get(tk, None)
```

also we probably need @kapilt to chime in with his opinion on doing this at all.  I think it is a nice quick solution to Azure parity for a useful feature but it is also a hack on a hack and unsure what GCP looks like here, so his call.

File: tools/c7n_azure/c7n_azure/filters.py [hunk 2]
```diff
@@ -28,27 +27,15 @@ class MetricFilter(Filter):
...
     schema = {
         'type': 'object',
         'required': ['type', 'metric', 'op', 'threshold'],
         'properties': {
             'metric': {'type': 'string'},
-            'op': {'enum': list(ops.keys())},
+            'op': {'enum': [
+                'eq', 'equal', 'ne', 'not-equal', 'gt', 'greater-than', 'ge',
+                'gte', 'le', 'lte', 'lt', 'less-than'
+            ]},
```
Comment (stefangordon) [inline tools/c7n_azure/c7n_azure/filters.py]: Seems worse?  `'op': {'enum': list(OPERATORS.keys())}}}` 

Were you having problems with docs on it? or?

File: c7n/filters/core.py (hunk 1 above)
Comment (stefangordon) [inline c7n/filters/core.py]:
```
            elif 'tags' in i:
                r = i.get("tags", {}).get(tk, None)
```

File: tools/c7n_azure/c7n_azure/filters.py (hunk 2 above)
Comment (tbarlow12) [reply tools/c7n_azure/c7n_azure/filters.py]: That was the original approach, but some of the keys in `OPERATORS` didn't seem to make a whole lot of sense.

File: c7n/filters/core.py (hunk 1 above)
Comment (tbarlow12) [reply c7n/filters/core.py]: Just to provide more detail on @stefangordon's last comment, in the scenario where the policy is something like
```
filters:
  - "tag:CreatedBy" : present
```
I created an `AzureValueFilter(ValueFilter)` with an overridden `get_resource_value` method, but I wasn't able to get to the overridden `get_resource_value` without the `type` specified. Went straight to the original `ValueFilter`.

File: tools/c7n_azure/c7n_azure/filters.py (hunk 2 above)
Comment (tbarlow12) [reply tools/c7n_azure/c7n_azure/filters.py]: And I switched it from using an in-class `operators` to `OPERATORS` as defined in `c7n.filters.core`

File: c7n/filters/core.py (hunk 1 above)
Comment (stefangordon) [reply c7n/filters/core.py]: Yeah would be trickier to change the default filter out I think.

File: tools/c7n_azure/c7n_azure/filters.py (hunk 2 above)
Comment (stefangordon) [reply tools/c7n_azure/c7n_azure/filters.py]: Oh! I see now.  I'm not sure I see it as better than what you had though?  Decoupling the schema from the list there might make maintaining it harder if that source list changes, etc.  I'd probably leave it like you had it before, or use the new list and let all the operators in, unless there is some other benefit I missed beyond avoiding the semi-duplicate operator map (not a problem I think).

File: c7n/filters/core.py (hunk 1 above)
Comment (kapilt) [reply c7n/filters/core.py]: Fwiw, the value filter is ripe for some refactoring, i was checking on out cyclomatic complexity scores on the code base, and this particular class is lights up as overly high.  I'll have a deeper look at this pr/use case later today.

File: tools/c7n_azure/c7n_azure/filters.py (hunk 2 above)
Comment (tbarlow12) [reply tools/c7n_azure/c7n_azure/filters.py]: 👍 I'll switch it back to how it was before

File: c7n/filters/core.py (hunk 1 above)
Comment (stefangordon) [reply c7n/filters/core.py]: Ok that makes sense - @tbarlow12 let's try to optimize that implementation per my first comment before we commit though, we don't want to implement our own `.get` as I'm sure it is slower than the real thing.

**Pull request merged**